### PR TITLE
[Store onboarding] Setting to show/hide the list

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #9561

- [ ] ⚠️ Please review https://github.com/woocommerce/woocommerce-ios/pull/9572 as well

## Description

We want to allow the user to hide/dismiss the Store onboarding task list.

This PR,

- Adds a setting to hide or show the onboarding list from the settings screen.
- Adds tracking event.

## Testing instructions

- Enable `hideStoreOnboardingTaskList` feature flag by returning `true` from `DefaultFeatureFlagService`
- Login into a store which has pending store onboarding tasks
- Tap `Menu` -> `Settings`
- Toggle the `Store Setup List` switch and validate that the store onboarding task list in the dashboard is shown/hidden accordingly. 
- Hiding the list from the dashboard by tapping `...` -> `Hide store setup list` should update the settings screen as well.

**Tracks**

Settings
- Tap `Menu` -> `Settings`
- Toggle the `Store Setup List` switch from ON to OFF.
- Validate that the following event is tracked.

```
Tracked store_onboarding_hide_list, properties: [AnyHashable("blog_id"): 218298555, AnyHashable("is_wpcom_store"): false, AnyHashable("pending_tasks"): "payments,store_details", AnyHashable("source"): "settings"]
```

Dashboard
- From the dashboard hide the task list by tapping `...` -> `Hide store setup list`
- Validate that the following event is tracked.

```
Tracked store_onboarding_hide_list, properties: [AnyHashable("pending_tasks"): "payments,store_details", AnyHashable("is_wpcom_store"): false, AnyHashable("blog_id"): 218298555, AnyHashable("source"): "onboarding_list"]
```

Validate that the  `pending_tasks` value matches the pending onboarding tasks. 

## Screenshots

Simulator Screen Recording - iPhone 14 - 2023-04-28 at 07 38 08

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Analytics/WooAnalyticsEvent+StoreOnboarding.swift [hunk 1]
```diff
@@ -2,26 +2,41 @@ import struct Yosemite.StoreOnboardingTask
...
+                              properties: [Key.source: source.rawValue,
+                                           Key.pendingTasks: pendingTasks.map({ $0.analyticsValue }).sorted().joined(separator: ",")])
+        }
+    }
+}
+
+extension WooAnalyticsEvent.StoreOnboarding {
+    enum Source: String {
+        case onboardingList = "onboarding_list"
+        case settings = "settings"
```
Comment (itsmeichigo) [inline WooCommerce/Classes/Analytics/WooAnalyticsEvent+StoreOnboarding.swift]: Nit: we can omit the value here since it matches the case.

**Review by itsmeichigo — APPROVED** [review]
LGTM 👍 

❓ Do we want to enable the feature flag to release this for the next version?

**Comment by selanthiraiyan** [comment]
> Do we want to enable the feature flag to release this for the next version?

There is a discussion about the `pending_tasks` property of the tracks event. I will enable the feature once we decide it. 
p1682649959151279/1681901420.031229-slack-C045CUK1Y3U

File: WooCommerce/Classes/Analytics/WooAnalyticsEvent+StoreOnboarding.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/Analytics/WooAnalyticsEvent+StoreOnboarding.swift]: Done in https://github.com/woocommerce/woocommerce-ios/pull/9573/commits/9742a9f4156b4d2c4129e2194177dbd214b17faa

**Pull request merged**